---
title: "Resources"
output:
  html_document:
    df_print: paged
    toc: true
    toc_float: true
---

The internet has an infinite number of resources available - if you can google than you can code! 

Don't forget about youtube - an excellent resources for those that are visual learners.

Below are just _some_ of the resources available, mainly tailored around genomics, however the Data and Software Carpentry resources are an excellent starting point for general information.

***
***

# Where to start?

It can be a bit daunting knowing where to start. There are so many programs, packages, applications, tutorials - it's important to remember you cannot do them all!

My personal advice (remember I am mac/linux user) is to start my learning how to do commandline (i.e. terminal) and know some basics of the `bash` language.
Once you can perform some basic commands then move on to learning R/RStudio.

My reason for recommending RStudio is it will save you lots of time in the long term.
The RStudio community is huge and everything is free so you never have to worry about no longer having access.
There is a wealth of packages for RStudio you can do everything, the whole workflow - data entry, manipulating and summarising spreadsheets, statistical analysis, data visualisation and then writing your report/thesis/publication etc.
It will save you long term as you won't have to learn so many additional programs (i.e. mapping programs, photoshop/figure editor software, statistics, the list goes on...).
Most often people will pick and choose to do certain elements in RStudio and then use programs they are familar with for the other bits....the good news is you can take it at your own pace.

**Siobhon's recommendations for beginners**

* [Software Carpentry](https://software-carpentry.org/)
    + [The Unix Shell](http://swcarpentry.github.io/shell-novice/) - **REQUIRED** (this is an essential start for beginners!)
    + [Version Control with Git](http://swcarpentry.github.io/git-novice) - this provides a platform for you to keep track and share your code. I suggest you come back to this later [desktop version](https://desktop.github.com/)
    + [Programming with Python](http://swcarpentry.github.io/python-novice-inflammation) - come back to python, this is another scripting language. A popular one among many coders, but often for bioinformatics most programs use `bash` (the one you learnt in 'The Unix Shell' tutorial)
    + [Programming with R](http://swcarpentry.github.io/r-novice-inflammation/) - **HIGHLY RECOMMENDED**
    + [R for reproducible scientific analysis](http://swcarpentry.github.io/r-novice-gapminder/) - recommended extra, although you can probably skip some lessons
<br></br>

Even if you think your field is not ecology, trust me this is an excellent place to start learning RStudio.

* [Data Carpentry](http://www.datacarpentry.org/lessons/)
    + Ecology Lessons
        - [Data organisation in spreadsheets](https://datacarpentry.org/spreadsheet-ecology-lesson/) - **HIGHLY RECOMMENDED**
        - [Data analysis and visualization in R](https://datacarpentry.org/R-ecology-lesson/) - **REQUIRED**

While you maybe tempted to look at the Genomics lessons (of course go ahead if you have a burning desire!) they are still very much a work in progress, and I would not recommend them at this stage. 
 
***

# R/RStudio 

In addition to the above [software carpentry](https://software-carpentry.org/lessons/index.html) and data [carpentry](https://datacarpentry.org/) resources some additional places to start include:

* [Swirl](http://swirlstats.com/) is a great interactive tutorials that run directly in R. Useful for statistical analysis and basic functions.
* [R Tutorial](http://www.r-tutor.com/r-introduction) offers here a couple of introductory tutorials on basic R concepts.
* [Code School](http://tryr.codeschool.com/) provides some more tutorials on basic R syntax and basics.
* [Quick-R](https://www.statmethods.net/r-tutorial/index.html) is great for tutorials on statistical analysis.
* [STAT 457](https://stat545.stat.ubc.ca/) course website for the 2019-2020 edition of STAT 545A and STAT 547M, colloquially known as just “STAT 545”, delivered at The University of British Columbia in Vancouver, BC.

**R Cheat Sheets**
There are a number of cheatsheets and other reference documenation available for R.

* A link to the offical R cheat sheets is [here](https://www.rstudio.com/resources/cheatsheets/).
* Other useful cheatsheets by DataCamp are available [here](https://www.datacamp.com/community/data-science-cheatsheets).
* Links to other useful resources availbale [here](http://www.datasciencefree.com/cheatsheets.html), [here](https://r-dir.com/reference/crib-sheets.html), [here](https://www.analyticsvidhya.com/blog/2017/02/top-28-cheat-sheets-for-machine-learning-data-science-probability-sql-big-data/) and [here](https://sites.google.com/site/rdatamining/docs/R-refcard-data-mining.pdf?attredirects=0)

***

# Sequence & Phylogenetics

More coming soon - see my personal page [here](https://siobhon-egan.github.io/bioinfo-phylo/index.html) for some inspiration in the mean time.

***

# NGS Sequence analysis

This pipeline makes use of [USEARCH](https://www.drive5.com/usearch/) for preprocessing and [QIIME2](https://qiime2.org/) for taxonomy assignment and analysis - both of these come with a number of different tutorials with example data for you to work through. I strongly suggest you work through these **critically** don't just copy and paste and press eneter (even thought it is tempting!) - make sure you understand what and why you are doing things. 

Other comparable alternatives

* [QIIME2](https://docs.qiime2.org/2018.2/plugins/) - although this webpage uses QIIME only from the taxonomy step onwards, pre-processing steps can be done in QIIME2 make using of various Plugins
* [Mothur](https://www.mothur.org/wiki/Analysis_examples) - a lot of information and resources available
* [VSEARCH](https://github.com/torognes/vsearch) - made as an alternative to USEARCH
* [FastQC](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/) - a common tool used for basic fastq commands and manipulation

Some other useful references and analysis tools for genomics

* [RNAseq analysis](https://bioinformatics-core-shared-training.github.io/RNAseq-R/) - a link to similar content of RNAseq by combine Australia is also available [here](http://combine-australia.github.io/RNAseq-R/) 
* [Next Gen Seek](http://nextgenseek.com)
* [Bits of DNA](http://liorpachter.wordpress.com/seq)
* [RNA-Seq Blog](http://www.rna-seqblog.com)
* [Journal of Next Generation Sequencing & Applications](http://www.omicsonline.org/next-generationsequencing-
applications.php)
* [CoreGenomics](http://core-genomics.blogspot.com)
* [Next-Gen Sequencing](http://nextgenseq.blogspot.com)
* [Omics! Omics!](http://omicsomics.blogspot.com)
* [In Between Lines of Code](http://flxlexblog.wordpress.com)
* [Kevin’s GATTACA World](http://kevin-gattaca.blogspot.com)
* [Blog @ Illumina](http://blog.illumina.com)
* [Next Generation Technologist](http://www.yuzuki.org)
* [Applied compuational genomics](https://github.com/quinlan-lab/applied-computational-genomics)
* [OSSU bioinformatics](https://github.com/ossu/bioinformatics)


## Microbial analysis

Microbial analysis has revieved the most attention with respect to amplicon based NGS approaches; however with some refining these pipelines can also be used for other amplicon based NGS studies (e.g. COI (metazoa), ITS (fungi), TrnL (plant), 18S (protozoa) just to name a few). Plus there is of course overlap with other genomics tools as well.
There can be alot of overlap in these so its up to you to see what works for you. Generally those in R allow more customisation however may be more difficult to grasp initially. GUI's are good to get a quick handle on your data, however you are usually limited in the customisation of the graphical outputs from these programs. Some of these are more of a stand alone package however the majority will utilise a common format (e.g. phyloseq objects are common throughout many of these). 

* [phyloseq](https://joey711.github.io/phyloseq/) - Analyze microbiome census data using R
* [microbiomeSeq](http://userweb.eng.gla.ac.uk/umer.ijaz/projects/microbiomeSeq_Tutorial.html#install) - An R package for microbial community analysis in an environmental context
* [metacoder](https://grunwaldlab.github.io/metacoder_documentation/) - An R package for metabarcoding research planning and analysis
* [microbiome R](http://microbiome.github.io/microbiome/) - Microbiome R package (extending phyloseq)
* [microbiomeutilities](https://microsud.github.io/microbiomeutilities/) - Extending and supporting package based on microbiome and phyloseq (currently in developmental stage)
* [R microbiota](https://rpubs.com/dillmcfarlan/R_microbiotaSOP) - Microbiota analysis in R
* [MicrobiobeMiseq](http://deneflab.github.io/MicrobeMiseq/) - Analyses of microbial community composition and diversity in R using phyloseq
* [Bioconductor Microbiome](https://bioconductor.org/help/course-materials/2017/BioC2017/Day1/Workshops/Microbiome/MicrobiomeWorkflowII.html) - Microbiome data analysis: from raw reads to community analysis
* [Ampvis2](https://microsud.github.io/Tools-Microbiome-Analysis/) - Tools for visualising amplicon data
* [dada2](https://benjjneb.github.io/dada2/index.html) - Fast and accurate sample inference from amplicon data with single-nucleotide resolution
* [mare](https://github.com/katrikorpela/mare) - Pipeline for microbiota analysis based on 16S-amplicon reads
* [metagenomeSeq](http://cbcb.umd.edu/software/metagenomeSeq/) - Statistical analysis for sparse high-throughput sequencing
* [Rhea](https://lagkouvardos.github.io/Rhea/) - A set of R scripts for the analysis of microbial profiles
* [taxize](https://github.com/ropensci/taxize) - A taxonomic toolbelt for R
* [LabDSV](http://ecology.msu.montana.edu/labdsv/R/) - Ordination and multivariate analysis for ecology
* [phylogeo](http://zachcp.github.io/phylogeo) - An R package for geographic analysis and visualization of microbiome data
* [qiimer](https://github.com/kylebittinger/qiimer) - R functions to read QIIME output files and create figures
* [RAM](https://rdrr.io/cran/RAM/) - R for amplicon-sequencing-based microbial-ecology

## RNASeq

...

## Database

Links to useful taxonomy databases go here...


***

# Epidemiology and Statistics

**Web/online resources**

* [ClinEpiDB](https://clinepidb.org/ce/app/) - Advancing global public health by facilitating the exploration and analysis of epidemiological studies. ClinEpiDB, launched in February 2018, is an open-access online resource enabling investigators to maximize the utility and reach of their data and to make optimal use of data released by others.
More coming soon
* [Epitool](https://epitools.ausvet.com.au/) - This site is developed and maintained by [Ausvet](https://www.ausvet.com.au/). The site is intended for use by epidemiologists and researchers involved in estimating disease prevalence or demonstrating freedom from disease through structured surveys, or in other epidemiological applications.
* [VassarStats](http://vassarstats.net/) - A useful and user-friendly tool for performing statistical computation (probabilities, regression, t-test, ANOVA and more)

***

# RMarkdown

More coming soon


## Writing your thesis

...

## Making a webpage

...

***

# Mapping

More coming soon

***

# Online modules and courses    
There are a number of online modules and courses are avilable too, some of which you have to pay for but there are many free resources available too.

Popular sites include:

* [Courses.edx](https://courses.edx.org)
    + [DNA sequencing](https://www.coursera.org/learn/dna-sequencing) - the lessons labelled 'practical' are particularly useful as they have split screen and run through the code line by line. Although they do not use amplicon based NGS examples they do inlclude good fundamental lessons (including how to read a fastq file)
    + [Bioinformatics methods 1](https://www.coursera.org/learn/bioinformatics-methods-1) - while it does not have recorded practical it goes have a 'post-lab lecture' which is helpful. Their handouts are very detailed, however again they do not give examples specific to NGS but more general bioinformatics.
*	[Coursera](https://www.coursera.org) - a whealth of courses to make your way through
*	[Datacamp](https://www.datacamp.com/home) - a fantastic resource for data science and anything coding related. The "on-screen" terminal has some pros and cons to it, give the free lessons ago (there is more than enough to keep you busy) and see how you like i - just make sure you take good notes!


