<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Diversity and Phylogeny Analysis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">My Website</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="resources.html">Resources</a>
</li>
<li>
  <a href="usearch-analysis.html">USEARCH Analysis</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    QIIME2
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="QIIME2-NativeInstall.html">Install</a>
    </li>
    <li>
      <a href="QIIME2-DataImport.html">Data Import</a>
    </li>
    <li>
      <a href="QIIME2-DiversityPhylogeny.html">Diversity and phylogeny</a>
    </li>
    <li>
      <a href="QIIME2-Taxonomy.html">Taxonomy</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Diversity and Phylogeny Analysis</h1>

</div>


<p>This workflow follows documentation from QIIME2 documents on <a href="√">tutorials</a> - mainly the moving pictures tutorial.</p>
<p><strong>16S amplicon NGS analysis</strong></p>
<p>This notebook continues on from the notebook on data import &amp; preliminary analysis, and native installation of QIIME2 following the USEARCH pipeline.</p>
<div id="generate-a-tree-for-phylogenetic-diversity-analyses" class="section level3">
<h3>Generate a tree for phylogenetic diversity analyses</h3>
<p>QIIME supports several phylogenetic diversity metrics, including Faith’s Phylogenetic Diversity and weighted and unweighted UniFrac. In addition to counts of features per sample (i.e., the data in the <code>FeatureTable[Frequency]</code> QIIME 2 artifact), these metrics require a rooted phylogenetic tree relating the features to one another. This information will be stored in a <code>Phylogeny[Rooted]</code> QIIME 2 artifact. The following steps will generate this QIIME 2 artifact.</p>
<p>First, we perform a multiple sequence alignment of the sequences in our <code>FeatureData[Sequence]</code> to create a <code>FeatureData[AlignedSequence]</code> QIIME 2 artifact. Here we do this with the mafft program.</p>
<pre class="python"><code>qiime alignment mafft \
--i-sequences sequences.qza \
--o-alignment aligned-sequences.qza</code></pre>
<p><strong>Output artifacts:</strong></p>
<pre class="python"><code>aligned-sequences.qza</code></pre>
<p>Next, we mask (or filter) the alignment to remove positions that are highly variable. These positions are generally considered to add noise to a resulting phylogenetic tree.</p>
<pre class="python"><code>qiime alignment mask \
--i-alignment aligned-sequences.qza \
--o-masked-alignment masked-aligned-sequences.qza</code></pre>
<p><strong>Output artifacts:</strong></p>
<pre class="python"><code>masked-aligned-sequences.qza</code></pre>
<p>Next, we’ll apply FastTree to generate a phylogenetic tree from the masked alignment.</p>
<pre class="python"><code>qiime phylogeny fasttree \
--i-alignment masked-aligned-sequences.qza \
--o-tree unrooted-tree.qza</code></pre>
<p><strong>Output artifacts</strong></p>
<pre class="python"><code>unrooted-tree.qza</code></pre>
<p>The FastTree program creates an unrooted tree, so in the final step in this section we apply midpoint rooting to place the root of the tree at the midpoint of the longest tip-to-tip distance in the unrooted tree.</p>
<pre class="python"><code>qiime phylogeny midpoint-root \
--i-tree unrooted-tree.qza \
--o-rooted-tree rooted-tree.qza</code></pre>
<p><strong>Output artifacts</strong></p>
<pre class="python"><code>rooted-tree.qza</code></pre>
<p><strong>Exporting trees</strong></p>
<p>Currently QIIME2 does not have the function to visualize trees, so you will need to export the tree and visualize using your own program.</p>
<p>To export the tree:</p>
<pre class="python"><code>qiime tools export \
  unrooted-tree.qza \
  --output-dir exported-unrooted-tree/
  
qiime tools export \
  rooted-tree.qza \
  --output-dir exported-rooted-tree/</code></pre>
<p>A <code>Phylogeny[Unrooted]</code> artifact will be exported as a newick formatted file.</p>
</div>
<div id="alpha-and-beta-diversity-analysis" class="section level3">
<h3>Alpha and beta diversity analysis</h3>
<p>QIIME 2’s diversity analyses are available through the <code>q2-diversity</code> plugin, which supports computing alpha and beta diversity metrics, applying related statistical tests, and generating interactive visualizations. We’ll first apply the core-metrics-phylogenetic method, which rarefies a <code>FeatureTable[Frequency]</code> to a user-specified depth, computes several alpha and beta diversity metrics, and generates principle coordinates analysis (PCoA) plots using Emperor for each of the beta diversity metrics. The metrics computed by default are:</p>
<ul>
<li>Alpha diversity
<ul>
<li>Shannon’s diversity index (a quantitative measure of community richness)</li>
<li>Observed OTUs (a qualitative measure of community richness)</li>
<li>Faith’s Phylogenetic Diversity (a qualitiative measure of community richness that incorporates phylogenetic relationships between the features)</li>
<li>Evenness (or Pielou’s Evenness; a measure of community evenness)</li>
</ul></li>
<li>Beta diversity
<ul>
<li>Jaccard distance (a qualitative measure of community dissimilarity)</li>
<li>Bray-Curtis distance (a quantitative measure of community dissimilarity)</li>
<li>unweighted UniFrac distance (a qualitative measure of community dissimilarity that incorporates phylogenetic relationships between the features)</li>
<li>weighted UniFrac distance (a quantitative measure of community dissimilarity that incorporates phylogenetic relationships between the features)</li>
</ul></li>
</ul>
<p>An important parameter that needs to be provided to this script is <code>--p-sampling-depth</code>, which is the even sampling (i.e. rarefaction) depth. Because most diversity metrics are sensitive to different sampling depths across different samples, this script will randomly subsample the counts from each sample to the value provided for this parameter. For example, if you provide <code>--p-sampling-depth 500</code>, this step will subsample the counts in each sample without replacement so that each sample in the resulting table has a total count of <code>500</code>. If the total count for any sample(s) are smaller than this value, those samples will be dropped from the diversity analysis. Choosing this value is tricky. We recommend making your choice by reviewing the information presented in the <code>table.qzv</code> file that was created above and choosing a value that is as high as possible (so you retain more sequences per sample) while excluding as few samples as possible.</p>
<div class="alert alert-success">
<strong>Question</strong> View the table.qzv QIIME 2 artifact, and in particular the Interactive Sample Detail tab in that visualization. What value would you choose to pass for <code>--p-sampling-depth?</code> How many samples will be excluded from your analysis based on this choice? How many total sequences will you be analyzing in the core-metrics-phylogenetic command?
</div>
<pre class="python"><code>qiime diversity core-metrics-phylogenetic \
--i-phylogeny rooted-tree.qza \
--i-table feature-table-1.qza \
--p-sampling-depth 500 \
--m-metadata-file sample-metadata.tsv \
--output-dir core-metrics-results</code></pre>
<p><strong>Output artifacts:</strong></p>
<pre class="python"><code>- core-metrics-results/bray_curtis_distance_matrix.qza
- core-metrics-results/bray_curtis_pcoa_results.qza
- core-metrics-results/evenness_vector.qza
- core-metrics-results/faith_pd_vector.qza
- core-metrics-results/jaccard_distance_matrix.qza
- core-metrics-results/jaccard_pcoa_results.qza
- core-metrics-results/observed_otus_vector.qza
- core-metrics-results/rarefied_table.qza
- core-metrics-results/shannon_vector.qza
- core-metrics-results/unweighted_unifrac_distance_matrix.qza
- core-metrics-results/unweighted_unifrac_pcoa_results.qza
- core-metrics-results/weighted_unifrac_distance_matrix.qza
- core-metrics-results/weighted_unifrac_pcoa_results.qza</code></pre>
<p><strong>Output visualizations:</strong></p>
<pre class="python"><code>- core-metrics-results/bray_curtis_emperor.qzv
- core-metrics-results/jaccard_emperor.qzv
- core-metrics-results/unweighted_unifrac_emperor.qzv
- core-metrics-results/weighted_unifrac_emperor.qzv</code></pre>
<p>After computing diversity metrics, we can begin to explore the microbial composition of the samples in the context of the sample metadata. This information is present in the sample metadata fie.</p>
<p>We’ll first test for associations between discrete metadata categories and alpha diversity data. We’ll do that here for the Faith Phylogenetic Diversity (a measure of community richness) and evenness metrics.</p>
<pre class="python"><code>qiime diversity alpha-group-significance \
--i-alpha-diversity core-metrics-results/faith_pd_vector.qza \
--m-metadata-file metadata.tsv \
--o-visualization core-metrics-results/faith-pd-group-significance.qzv

qiime diversity alpha-group-significance \
--i-alpha-diversity core-metrics-results/evenness_vector.qza \
--m-metadata-file metadata.tsv \
--o-visualization core-metrics-results/evenness-group-significance.qzv</code></pre>
<p><strong>Output visualizations:</strong></p>
<pre class="python"><code>core-metrics-results/evenness-group-significance.qzv
core-metrics-results/faith-pd-group-significance.qzv</code></pre>
<div class="alert alert-success">
<strong>Question</strong> What discrete sample metadata categories are most strongly associated with differences in microbial community richness? Are these differences statistically different?
</div>
<div class="alert alert-success">
<strong>Question</strong> What discrete sample metadata categories are most strongly associated with differences in microbial community evenness? Are these differences statistically different?
</div>
<p>If you have a continuous variable in your metadata (e.g., <code>DaysSinceExperimentStart</code>) then you can tested if these are correlated with alpha diversity. If you’re interested in performing those tests you can use the qiime diversity <code>alpha-correlation</code> command.</p>
<p>Next we’ll analyze sample composition in the context of discrete metadata using <code>PERMANOVA</code> (first described in Anderson (2001)) using the <code>beta-group-significance</code> command. The following commands will test whether distances between samples within a group, such as samples from the same host species, are more similar to each other then they are to samples from the hosts. If you call this command with the <code>--p-pairwise parameter</code>, as we’ll do here, it will also perform pairwise tests that will allow you to determine which specific pairs of groups (e.g., human vs dog hosts) differ from one another, if any. This command can be slow to run, especially when passing <code>--p-pairwise</code>, since it is based on permutation tests. So, unlike the previous commands, we’ll run this on specific categories of metadata that we’re interested in exploring, rather than all metadata categories that it’s applicable to. Here we’ll apply this to our unweighted UniFrac distances, using two sample metadata categories (<code>Host</code> and <code>TickSpecies</code>), as follows.</p>
<pre class="python"><code>qiime diversity beta-group-significance \
--i-distance-matrix core-metrics-results/unweighted_unifrac_distance_matrix.qza \
--m-metadata-file metadata.tsv \
--m-metadata-category Host \
--o-visualization core-metrics-results/unweighted-unifrac-Host-significance.qzv \
--p-pairwise

qiime diversity beta-group-significance \
--i-distance-matrix core-metrics-results/unweighted_unifrac_distance_matrix.qza \
--m-metadata-file metadata.tsv \
--m-metadata-category TickSpecies \
--o-visualization core-metrics-results/unweighted-unifrac-TickSpecies-group-significance.qzv \
--p-pairwise</code></pre>
<p><strong>Output visualizations:</strong></p>
<pre class="python"><code>core-metrics-results/unweighted-unifrac-body-site-significance.qzv
core-metrics-results/unweighted-unifrac-subject-group-significance.qzv</code></pre>
<div class="alert alert-success">
<strong>Question</strong> Are the associations between Tick species and differences in microbial composition statistically significant? How about hosts? What specific hosts/tick species are significantly different from each other?
</div>
<p>Again, none of the continuous sample metadata that we have for this data set are correlated with sample composition, so we won’t test for those associations here. If you’re interested in performing those tests, you can use the <code>qiime metadata distance-matrix</code> in combination with <code>qiime diversity mantel</code> and<code>qiime diversity bioenv</code> commands.</p>
<p>Finally, ordination is a popular approach for exploring microbial community composition in the context of sample metadata. QIIME uses the <a href="http://biocore.github.io/emperor/">Emperor tool</a> to explore principal coordinates (PCoA) plots in the context of sample metadata. While the <code>core-metrics-phylogenetic</code> command did already generate some Emperor plots, we want to pass an optional parameter, <code>--p-custom-axis</code>, which is very useful for exploring time series data. The PCoA results that were used in <code>core-metrics-phylogeny</code> are also available, making it easy to generate new visualizations with Emperor. This will generate Emperor plots for unweighted UniFrac and Bray-Curtis so that the resulting plot will contain axes for principal coordinate 1, principal coordinate 2, and days since the experiment start. We will use that last axis to explore how these samples changed over time <code>- DaysSinceAttachment</code>.</p>
<pre class="python"><code>qiime emperor plot \
--i-pcoa core-metrics-results/unweighted_unifrac_pcoa_results.qza \
--m-metadata-file metadata.tsv \
--p-custom-axis DaysSinceAttachment \
--o-visualization core-metrics-results/unweighted-unifrac-emperor-DaysSinceAttachment.qzv

qiime emperor plot \
--i-pcoa core-metrics-results/bray_curtis_pcoa_results.qza \
--m-metadata-file metadata.tsv \
--p-custom-axis DaysSinceAttachment \
--o-visualization core-metrics-results/bray-curtis-emperor-DaysSinceAttachment.qzv</code></pre>
<p><strong>Output visualizations:</strong></p>
<pre class="python"><code>core-metrics-results/bray-curtis-emperor-DaysSinceExperimentStart.qzv
core-metrics-results/unweighted-unifrac-emperor-DaysSinceExperimentStart.qzv</code></pre>
<div class="alert alert-success">
<strong>Question</strong> Do the Emperor plots support the other beta diversity analyses we’ve performed here? (Hint: Experiment with coloring points by different metadata.)
</div>
<div class="alert alert-success">
<strong>Question </strong> What differences do you observe between the unweighted UniFrac and Bray-Curtis PCoA plots?
</div>
</div>
<div id="alpha-rarefaction-plotting" class="section level3">
<h3>Alpha rarefaction plotting</h3>
<p>In this section we’ll explore alpha diversity as a function of sampling depth using the qiime diversity alpha-rarefaction visualizer. This visualizer computes one or more alpha diversity metrics at multiple sampling depths, in steps between 1 (optionally controlled with <code>--p-min-depth)</code> and the value provided as <code>--p-max-depth.</code> At each sampling depth step, 10 rarefied tables will be generated, and the diversity metrics will be computed for all samples in the tables. The number of iterations (rarefied tables computed at each sampling depth) can be controlled with <code>--p-iterations</code>. Average diversity values will be plotted for each sample at each even sampling depth, and samples can be grouped based on metadata in the resulting visualization if sample metadata is provided with the <code>--m-metadata-file</code> parameter.</p>
<pre class="python"><code>qiime diversity alpha-rarefaction \
--i-table feature-table-1.qza \
--i-phylogeny rooted-tree.qza \
--p-max-depth 10000 \
--m-metadata-file metadata.tsv \
--o-visualization alpha-rarefaction.qzv</code></pre>
<p><strong>Output visualizations:</strong></p>
<pre class="python"><code>- alpha-rarefaction.qzv</code></pre>
<p>The visualization will have two plots. The top plot is an alpha rarefaction plot, and is primarily used to determine if the richness of the samples has been fully observed or sequenced. If the lines in the plot appear to “level out” (i.e., approach a slope of zero) at some sampling depth along the x-axis, that suggests that collecting additional sequences beyond that sampling depth would not be likely to result in the observation of additional features. If the lines in a plot don’t level out, this may be because the richness of the samples hasn’t been fully observed yet (because too few sequences were collected), or it could be an indicator that a lot of sequencing error remains in the data (which is being mistaken for novel diversity).</p>
<p>The bottom plot in this visualization is important when grouping samples by metadata. It illustrates the number of samples that remain in each group when the feature table is rarefied to each sampling depth. If a given sampling depth d is larger than the total frequency of a sample s (i.e., the number of sequences that were obtained for sample s), it is not possible to compute the diversity metric for sample s at <code>sampling depth d</code>. If many of the samples in a group have lower total frequencies than d, the average diversity presented for that group at d in the top plot will be unreliable because it will have been computed on relatively few samples. When grouping samples by metadata, it is therefore essential to look at the bottom plot to ensure that the data presented in the top plot is reliable.</p>
<div class="alert alert-success">
<strong>Question</strong> The value that you provide for <code>--p-max-depth</code> should be determined by reviewing the <code>Frequency per sample</code> information presented in the <code>table.qzv</code> file that was created above. In general, choosing a value that is somewhere around the median frequency seems to work well, but you may want to increase that value if the lines in the resulting rarefaction plot don’t appear to be leveling out, or decrease that value if you seem to be losing many of your samples due to low total frequencies closer to the minimum sampling depth than the maximum sampling depth.
</div>
<div class="alert alert-success">
<strong>Question</strong> When grouping samples by Host and viewing the alpha rarefaction plot for the <code>observed_otus</code> metric, which Hosts (if any) appear to exhibit sufficient diversity coverage (i.e., their rarefaction curves level off)? How many sequence variants appear to be present in those Hosts?
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
