<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Beta Diversity and PCoA</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />
<link href="site_libs/ionicons-2.0.1/css/ionicons.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">My Website</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="about.html">
    <span class="ion ion-person"></span>
     
    About
  </a>
</li>
<li>
  <a href="resources.html">
    <span class="fa fa-book"></span>
     
    Resources
  </a>
</li>
<li>
  <a href="usearch-analysis.html">
    <span class="fa fa-laptop"></span>
     
    USEARCH Analysis
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    QIIME2
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="QIIME2-NativeInstall.html">Install</a>
    </li>
    <li>
      <a href="QIIME2-DataImport.html">Data Import</a>
    </li>
    <li>
      <a href="QIIME2-DiversityPhylogeny.html">Diversity and phylogeny</a>
    </li>
    <li>
      <a href="QIIME2-Taxonomy.html">Taxonomy</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    R analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="R-PhyloseqIntro.html">Phyloseq Intro</a>
    </li>
    <li>
      <a href="R-SubsettingData.html">Exploring Data</a>
    </li>
    <li>
      <a href="R-AlphaDiversity.html">Alpha Diversity &amp; Seq depth</a>
    </li>
    <li>
      <a href="R-MicrobialComposition.html">Microbial Composition</a>
    </li>
    <li>
      <a href="R-BetaPCoA.html">Beta Diversity and PCoA</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Beta Diversity and PCoA</h1>

</div>


<p>There are many different options when it comes to beta diversity analayis and plot ordination. The phyloseq documentation is a good place to start to understand the different types of analysis. You can also use the help function in r <code>?ordinate</code> and <code>?plot_ordination</code>.</p>
<p>The operation of the plot_ordination function also depends a lot on the: <a href="https://joey711.github.io/phyloseq/distance.html">Distance</a> and <a href="https://joey711.github.io/phyloseq/ordinate.html">Ordinate</a> functions.</p>
<p>The <code>subset_ord_plot</code> function is a “convenience function” intended to make it easier to retrieve a plot-derived <code>data.frame</code> with a subset of points according to a <code>threshold</code> and <code>method</code>. The meaning of the <code>threshold</code> depends upon the <code>method</code>. Phyloseq examples are avilable <a href="http://joey711.github.io/phyloseq/subset_ord_plot-examples">here</a>.</p>
<p>In this workflow we will use the plot ordinate feature to determine what variables affect the diversity of the tick microbiome.</p>
<div id="mds-on-bray-distances" class="section level3">
<h3>MDS on Bray Distances</h3>
<pre class="r"><code>ord_MDS &lt;- ordinate(samples_only, &quot;MDS&quot;, &quot;bray&quot;)
p &lt;- plot_ordination(samples_only, ord_MDS, type = &quot;samples&quot;, color = &quot;Genus_Species&quot;, shape = &quot;Tick_Instar&quot;, title = &quot;MDS ordination&quot;) + geom_point(size = 1)
print(p)</code></pre>
<div class="figure">
<img src="images/MDS-ord.png" />

</div>
<p>To facet the figure by a variable use ggplot command <code>facet_wrap</code></p>
<pre class="r"><code>p + facet_wrap(~Genus_Species)</code></pre>
<div class="figure">
<img src="images/MDS-ord-facet.png" />

</div>
<p>Another way that is useful to group samples is to use ggplot geom_polygon feature, however this is best used when there are only a few samples to plot and all samples of the same variable group closely together.</p>
<pre class="r"><code>p2 &lt;- plot_ordination(samples_only, ord_MDS, type=&quot;samples&quot;, color=&quot;Genus_Species&quot;)
p2 + geom_polygon(aes(fill=Genus_Species)) + geom_point(size=1) + ggtitle(&quot;Genus_Species&quot;)</code></pre>
<hr />
</div>
<div id="mds-pcoa-on-unifrac-distances" class="section level3">
<h3>MDS (“PCoA”) on Unifrac Distances</h3>
<p>Use the ordinate function to simultaneously perform weightd UniFrac and then perform a Principal Coordinate Analysis on that distance matrix (first line). Next pass that data and the ordination results to plot_ordination to create the ggplot2 output graphic with default ggplot2 settings.</p>
<p>The first line of code to ordinate the data can take a long time for large datasets</p>
<pre class="r"><code>ord_PCoA_unifrac = ordinate(samples_only, &quot;PCoA&quot;, &quot;unifrac&quot;, weighted=TRUE)
plot_ordination(samples_only, ordu, color=&quot;Genus_Species&quot;, shape=&quot;Tick_Instar&quot;)</code></pre>
</div>
<div id="heatmap-of-ordination" class="section level3">
<h3>Heatmap of ordination</h3>
<p>We can also present the ordination methods as a heat map. This works best for smaller datasets so we will subset the large <code>physeq1</code> data to only <em>Ixodes tasmani</em></p>
<pre class="r"><code>tasmani &lt;- subset_samples(physeq1, Tick_Species== &quot;tasmani&quot;)</code></pre>
<p>plot_heatmap(gpac, “DCA”, “none”, “SampleType”, “Family”)</p>
</div>

<p>Vector and Waterborne Pathogens Research Group. 2018. S Egan.</p>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
