<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Subsetting Data</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />
<link href="site_libs/ionicons-2.0.1/css/ionicons.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">My Website</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="about.html">
    <span class="ion ion-person"></span>
     
    About
  </a>
</li>
<li>
  <a href="resources.html">Resources</a>
</li>
<li>
  <a href="usearch-analysis.html">USEARCH Analysis</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    QIIME2
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="QIIME2-NativeInstall.html">Install</a>
    </li>
    <li>
      <a href="QIIME2-DataImport.html">Data Import</a>
    </li>
    <li>
      <a href="QIIME2-DiversityPhylogeny.html">Diversity and phylogeny</a>
    </li>
    <li>
      <a href="QIIME2-Taxonomy.html">Taxonomy</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    R analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="R-PhyloseqIntro.html">Phyloseq Intro</a>
    </li>
    <li>
      <a href="R-SubsettingData.html">Exploring Data</a>
    </li>
    <li>
      <a href="R-AlphaDiversity.html">Alpha Diversity &amp; Seq depth</a>
    </li>
    <li>
      <a href="R-MicrobialComposition.html">Microbial Composition</a>
    </li>
    <li>
      <a href="R-BetaPCoA.html">Beta Diversity and PCoA</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Subsetting Data</h1>

</div>


<div id="pre-processing-data" class="section level3">
<h3>Pre-processing data</h3>
<p>The phyloseq package also includes functions for filtering, subsetting, and merging abundance data. Filtering in phyloseq is designed in a modular fashion similar to the approach in the genefilter package. This includes the prune_taxa and prune_samples methods for directly removing unwanted indices, as well as the filterfun_sample and genefilter_sample functions for building arbitrarily complex sample-wise filtering criteria, and the filter_taxa function for taxa-wise filtering. You can try experimenting with different combinations of these functions. For example you may want to filter by sample, then by taxa. Use the basic commands functions to explore the different in the sub-sets.</p>
<p>Sub-set samples - exlude controls You will likely want to subset your dataset so that you only analyse your samples, i.e. exclude any PCR or extraction controls. Of course I recommend you check these samples first to ensure there is not important taxa in these samples. For example in the case of tick microbiome, we know there are certain tick-associated bacteria that we do not want to see in our control samples - otherwise this would indicate contamination.</p>
<hr />
<p><strong>Subset your samples (i.e. exclude controls)</strong></p>
<pre class="r"><code>samples_only &lt;- subset_samples(physeq1, Sample_Type == &quot;Sample&quot;)</code></pre>
<p><strong>Subset controls</strong></p>
<pre class="r"><code>controls &lt;- subset_samples(physeq1, Sample_Type == &quot;Control&quot;)</code></pre>
<p>Check the number of samples in your sample_only object compared to your entire physeq1 object.</p>
<pre class="r"><code>nsamples(physeq)
nsamples(samples_only)
nsamples(control)</code></pre>
<hr />
</div>
<div id="transforming-and-filtering-data" class="section level3">
<h3>Transforming and filtering data</h3>
<p>In the following example, the <code>Physeq1</code> object is first transformed to relative abundance, creating the new <code>PHr</code> object, which is then filtered (<code>PHfr</code> object) such that only OTUs with a mean greater than 10^-5 are kept.</p>
<pre class="r"><code>PHr  = transform_sample_counts(physeq1, function(x) x / sum(x) )
PHfr = filter_taxa(PHr, function(x) mean(x) &gt; 1e-5, TRUE)</code></pre>
<p>Remeber you can check the number of taxa at any stage by</p>
<pre class="r"><code>ntaxa(PHr)
ntaxa(PHfr)</code></pre>
<hr />
</div>
<div id="subsetting-data-by-taxa" class="section level3">
<h3>Subsetting data by taxa</h3>
<p>Here we can subset data based on taxa, and alter the threshold of abundance. This is particularly useful when you want to test what the threshold is for a “positive sample” in your NGS data.</p>
<p>Remember that the <code>Class==&quot;Taxa&quot;</code> needs to be spelt as it appears in the taxmat object. For example if you imported your taxaonomy file from Greengenes it is formatted like: <code>c__Taxa</code>, where the letter prefix refers to the heirachy. See the note under the importing of the taxonomy table.</p>
<pre class="r"><code>c_Alphaprob = subset_taxa(physeq1, Class==&quot;Alphaproteobacteria&quot;)
c_Alphaprob = prune_samples(sample_sums(c__Alphaprob)&gt;=20, c__Alphaprob)

g_Borrelia = subset_taxa(physeq1, Genus==&quot;Borrelia&quot;)
g_Borrelia = prune_samples(sample_sums(g_Borrelia)&gt;=1, g_Borrelia)

f_Bartonellaceae = subset_taxa(physeq1, Family==&quot;Bartonellaceae&quot;)
f_Bartonellaceae = prune_samples(sample_sums(f_Bartonellaceae)&gt;=1, f_Bartonellaceae)</code></pre>
<hr />
</div>
<div id="rarefying-data" class="section level3">
<h3>Rarefying data</h3>
<div class="alert alert-block alert-info">
Please note that the authors of phyloseq do not advocate using this as a normalization procedure, despite its recent popularity. See McMurdie and Holmes 2015 PLoS Comput. Biol. or <code>?rarefy_even_depth</code> in your R environment You will need to experiment with this line of code in order for it to make biological sense to your data set. We recommend reading the reference above about Rarefying microbial datasets.
</div>
<pre class="r"><code>physeq1.rarified &lt;- rarefy_even_depth(physeq1)
ntaxa(physeq1.rarified)
nsamples(physeq1.rarified)
rm(physeq1.rarified)
physeq1.rarified &lt;- rarefy_even_depth(physeq1, rngseed = 1000)
ntaxa(physeq1.rarified)
nsamples(physeq1.rarified)</code></pre>
<hr />
</div>
<div id="sampling-depth" class="section level3">
<h3>Sampling depth</h3>
<p>Credit to <a href="http://deneflab.github.io/MicrobeMiseq/demos/mothur_2_phyloseq.html">MicrobMiseq github</a></p>
<p>Make a data frame with a column for the read counts of each sample</p>
<pre class="r"><code>sample_sum_df &lt;- data.frame(sum = sample_sums(samples_only))</code></pre>
<p>Create histogram of sample read counts</p>
<pre class="r"><code>ggplot(sample_sum_df, aes(x = sum)) + 
  geom_histogram(color = &quot;black&quot;, fill = &quot;indianred&quot;, binwidth = 2500) +
  ggtitle(&quot;Distribution of sample sequencing depth&quot;) + 
  xlab(&quot;Read counts&quot;) +
  theme(axis.title.y = element_blank())</code></pre>
<div class="figure">
<img src="data/sampling-depth.png" />

</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
