<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>USEARCH v11 Amplicon Analysis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />
<link href="site_libs/ionicons-2.0.1/css/ionicons.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<div class="container-fluid main-container">

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->





<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">My Website</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="about.html">
    <span class="ion ion-person"></span>
     
    About
  </a>
</li>
<li>
  <a href="resources.html">
    <span class="fa fa-book"></span>
     
    Resources
  </a>
</li>
<li>
  <a href="setup.html">
    <span class="fa fa-building"></span>
     
    Set up
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-binoculars"></span>
     
    USEARCH Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="usearch-getting-started.html">Getting started</a>
    </li>
    <li>
      <a href="usearch-v11.html">USEARCH v11</a>
    </li>
    <li>
      <a href="usearch-previous.html">USEARCH previous</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-puzzle-piece"></span>
     
    QIIME2
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="QIIME2-NativeInstall.html">Install</a>
    </li>
    <li>
      <a href="QIIME2-DataImport.html">Data Import</a>
    </li>
    <li>
      <a href="QIIME2-DiversityPhylogeny.html">Diversity and phylogeny</a>
    </li>
    <li>
      <a href="QIIME2-Taxonomy.html">Taxonomy</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-puzzle-piece"></span>
     
    R analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="R-PhyloseqIntro.html">Phyloseq Intro</a>
    </li>
    <li>
      <a href="R-SubsettingData.html">Exploring Data</a>
    </li>
    <li>
      <a href="R-AlphaDiversity.html">Alpha Diversity &amp; Seq depth</a>
    </li>
    <li>
      <a href="R-MicrobialComposition.html">Microbial Composition</a>
    </li>
    <li>
      <a href="R-BetaPCoA.html">Beta Diversity and PCoA</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">USEARCH v11 Amplicon Analysis</h1>

</div>

<div id="TOC">
<ul>
<li><a href="#introduction">Introduction</a></li>
<li><a href="#fastq-info">Fastq info</a></li>
<li><a href="#merge-pairs">Merge pairs</a></li>
<li><a href="#trim-primers">Trim primers</a></li>
<li><a href="#quality-control-and-removing-dimer-sequences">Quality control and removing dimer sequences</a></li>
<li><a href="#renaming-sequences">Renaming sequences</a></li>
<li><a href="#removing-low-abundant-sequences">Removing low abundant sequences</a></li>
<li><a href="#clustering">Clustering</a></li>
<li><a href="#what-next">What next?</a></li>
</ul>
</div>

<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p><strong>Input</strong></p>
<p>Raw <code>fastq</code> files from illumina NGS platforms.</p>
<p><strong>Output</strong></p>
<p>Complied OTU or ZOTU table (taxonomy not assigned) and final sequences in <code>fasta</code> format.</p>
<p><strong>Assumptions</strong></p>
<p>Working within MacOS environment - there are slight differences when working within a linux environment. If you are using a virtual machine or cloud computing than often you will need to work within a linux environment. The documentation for usearch is the same for MacOS and linux with the exception that you will need to download the linux versions of the softwear, however take note it is primarily written for MacOS and some features may not be available. Please provide feedback if functions are not working within a linux environment.</p>
<p>Downloaded USEARCH. This pipeline requires USEARCH v11. For smaller datasets (e.g. one MiSeq run) the 34-bit files are okay, however if undertaking analysis of large datasets you will need to purchase the 64-bit.</p>
<p><strong>Data type</strong></p>
<p>This analysis is for paired-end reads generated on the illumina MiSeq</p>
<p><strong>Using this pipeline</strong></p>
<p>I strongly suggest you take a small sub-set of your data though the pipeline outlined below step-by-step first to get an understanding of what each command is doing. Importantly it will also allow to explore the influence of the different variables on your dataset.</p>
<p>Once you are happy with the parameters and that the script is working you can use the full script to analyse all your samples. Open your favourite text editor and combine all the sections of your workflow into a document with the file extension <code>.sh</code>. For example <code>16S_usearchv11.sh</code></p>
<p>Simply edit the script with your parameters, and save it in a directory with your raw fastq files.</p>
<p>All sub-scripts and a complete version of the pipeline can be found on my github repo <a href="https://github.com/siobhon-egan/usearch_analysis">siobhon-egan/usearch_analysis</a>.</p>
<p>In a bash terminal navigate to your directory and perform the following to execute the script and save the terminal output to a text file.</p>
<pre class="bash"><code>cd /Users/name/Documents/NGS-Analysis
chmod +x 16S_usearchv11.sh
./16S_usearchv11.sh &gt; 16S-NGS-usearchv11-output.txt</code></pre>
<p>*Note: the code below assumes you have installed USEARCH v11, file name <code>usearch11</code> and it is executable (if not navigate to the file and type <code>chmod +x usearch11</code> in the command line).</p>
<hr />
</div>
<div id="fastq-info" class="section level2">
<h2>Fastq info</h2>
<p><a href="https://www.drive5.com/usearch/manual/cmd_fastx_info.html">fastq_info command</a></p>
<p>This commands gives a short summary report of the sequences in a <code>fasta</code> or <code>fastq</code> file. It is useful for a first check on what is in a new file. The report is written to the console and can be saved to a text file using the <code>-output</code> option</p>
<pre class="bash"><code>#!/bin/bash

raw_data=&quot;raw_data&quot;
read_summary=&quot;1.read_summary&quot;

mkdir $read_summary

 for fq in $raw_data/*R1*.fastq
 do
  usearch11 -fastx_info $fq -output $read_summary/1a_fwd_fastq_info.txt
 done

 for fq in $raw_data/*R2*.fastq
 do
  usearch11 -fastx_info $fq -output $read_summary/1b_rev_fastq_info.txt
 done</code></pre>
<p>To look at just one fasta or fastq files, for example inspect your concatenate sequences. The -secs 5 options is good when just wanting a snap shot of a large sequences file, however it can be deleted if you can it to scan the entire file.</p>
<pre class="bash"><code>usearch11 -fastx_info sequences.fastq -secs 5 -output reads_info.txt</code></pre>
<p><span style="color:red"><strong>IMPORTANT!</strong></span></p>
<p>At this point it is important to look at the ‘EE’ value which means expected error.</p>
<p>Detailed information on this can be found <a href="https://www.drive5.com/usearch/manual/exp_errs.html">here</a>.</p>
<p>In short you want your EE value to be under 2, usually it is lower for the forward reads than the reverse reads.</p>
<hr />
</div>
<div id="merge-pairs" class="section level2">
<h2>Merge pairs</h2>
<p><a href="https://www.drive5.com/usearch/manual/cmd_fastq_mergepairs.html">fastq_mergepairs command</a></p>
<p>This command merges paired-end reads to create consensus sequences and, optionally, consensus quality scores. This command has many features and options so I recommend spending some time browsing the documentation to get familiar with the capabilities of fastq_mergepairs and issues that arise in read merging.</p>
<p>As is the convention for illumina paried reads the forward sequences are labeled <code>samplename_SXX_L001_R1_001.fastq</code> and reverse sequences labeled <code>samplename_SXX_L001_R2_001.fastq</code>.</p>
<p>There are many additional options for this command that are specified <a href="https://www.drive5.com/usearch/manual/merge_options.html">here</a>, and a thorough outline of reviewing fastq_mergepairs reports <a href="https://www.drive5.com/usearch/manual/merge_report.html">here</a>.</p>
<pre class="bash"><code>#!/bin/bash

raw_data=&quot;raw_data&quot;
merged_data=&quot;2.merged_reads&quot;
# Maximum no of mismatches in the alignment - Default 5. Consider increasing if you have long overlaps.
maxdiffs=&quot;15&quot;
# Discard pair if alugnment is shorter than this value - Default 16 bp
overlap=&quot;50&quot;

mkdir ${merged_reads}
mkdir working1

#*****************************************************************************************
# Step1: merge data with usearch9 -fastq-filter

for file1 in ${raw_data}/*R1_001.fastq
    do

        echo &quot;&quot;
        echo ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        echo Merging paired reads
        echo forward reads are:
        echo $(basename ${file1})
        echo reverse reads are:
        echo $(basename ${file1} R1_001.fastq)R2_001.fastq

    usearch11 -fastq_mergepairs ${file1} -reverse &quot;${raw_data}/$(basename -s R1_001.fastq ${file1})R2_001.fastq&quot; -fastqout &quot;working1/$(basename &quot;$file1&quot;)&quot; -fastq_maxdiffs ${maxdiffs} -fastq_minovlen ${overlap} -report ${merged_data}/2a_merging_seqs_report.txt -tabbedout ${merged_data}/2b_tabbedout.txt
done

#*****************************************************************************************
# Step 2: Remove &quot;_L001_R1_001&quot; from filenames

for file2 in working1/*.fastq
    do

        rename=&quot;$(basename ${file2} _L001_R1_001.fastq).fastq&quot;

        mv ${file2} ${merged_data}/${rename}
done

#*****************************************************************************************
# Removing working directory

        rm -r working1</code></pre>
<div class="alert alert-block alert-info">
<strong> Note</strong> There are a number of other commands that you may find useful at this step available on the USEARCH documenation page here. After trials with 16S v1-2 amplicon NGS data I have found that in most cases altering these parameters make little to no difference to the outcome of the sequences (i.e % merged) and hence they are obmitted from this pipeline - however the two exceptions are <code>fastq_minovlen</code> and <code>fastq-maxdiffs</code>. Again it is important to remember the purpose of this line of code, it is simply to merge the forward and reverse sequences not to act as a quality control step.
</div>
<hr />
</div>
<div id="trim-primers" class="section level2">
<h2>Trim primers</h2>
<p><a href="https://drive5.com/usearch/manual/cmd_search_pcr2.html">search_pcr2 command</a></p>
<p>This command extracts segments <em>between</em> matches to a primer fair. Input can be <code>fasta</code> or <code>fastq</code>. Primer sequences must be in 5<code>-3</code> direction, and it does support <a href="https://drive5.com/usearch/manual/IUPAC_codes.html">IUPAC codes for sequence wildcard letter</a>. The <code>-strand</code> option must be specific, where <code>-strand plus</code> searches for hits on the forward stand only and <code>-strand both</code> searches for hits on both forward and reverse-complemented strands. The <code>-maxdiff</code> option specifies the maximum allowed number of mismatches with a primer, default = 2. See USEARCH documentation for more information.</p>
<p>It is recommended that primers be trimmed before quality filtering of <code>fastq</code> files, as every base can cause an increase in the <a href="https://drive5.com/usearch/manual/exp_errs.html">expected errors</a>.</p>
<p>A common alternative to using the <code>search_pcr2</code> command is the <code>fastx_truncate</code> <a href="https://drive5.com/usearch/manual/cmd_fastx_truncate.html">command</a>. Depending on your data you may like to make your own choice, however I feel searching for the primer set specifically yeilds a more reliable result.</p>
<pre class="bash"><code>#!/bin/bash

merged_data=&quot;2.merged_reads&quot;
primer_matched=&quot;3a.primer_matches&quot;
primer_not_matched=&quot;3b.primer_mismatches&quot;
# Eneter primer sequences 5`-3` (degenerate bases OK)
fwd_primer=&quot;AGAGTTTGATCCTGGCTYAG&quot; #16S v1-2 primer, ref Gofton et al. Parasites &amp; Vectors (2015) 8:345
rev_primer=&quot;TGCTGCCTCCCGTAGGAGT&quot; #16S v1-2 primer, ref Turner et al. J Eukaryot Microbiol (1999) 46(4):327-338

echo %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
echo Triming primers and distal bases
echo %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
echo &quot;&quot;


Creating working directories

mkdir ${primer_matched}
mkdir ${primer_not_matche

for file3 in ${merged_data}/*.fastq
    do

        echo &quot;&quot;
        echo ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        echo Trimming primers step 1: finding seqs with FWD primer
        echo input is:
        echo ${file3}

    usearch11 -search_pcr2 ${file3} -fwdprimer ${fwd_primer} \
    -revprimer ${rev_primer} \
    -strand both -fastqout &quot;${primer_matched}/$(basename ${file3})&quot; -notmatchedfq &quot;${primer_not_matched}/$(basename ${file3})&quot; -tabbedout pcr2_primermatches_output.txt
done</code></pre>
<hr />
</div>
<div id="quality-control-and-removing-dimer-sequences" class="section level2">
<h2>Quality control and removing dimer sequences</h2>
<p><a href="https://www.drive5.com/usearch/manual/cmd_fastq_filter.html">fastq_filter command</a></p>
<p>This command performs quality filtering and converts <code>fastq</code> files to <code>fasta</code> format. Again there are a number of filtering options available with this command that you can explore, however below is what I have found the most approriate for 16S amplicon NGS data tested so far.</p>
<pre class="bash"><code>#!/bin/bash

primer_matched=&quot;3a.primer_matches&quot;
QF_reads=&quot;4.quality_filtered&quot;
# Discard reads with &gt; E total expected errors for all bases in the read after any truncation options have been applied. Usually set at 1, however may want to lower to 0.5 or 0.25 for more stringent filtering.
max_ee=&quot;1&quot;
# Enter min length of sequence for trimming in bp (eg. to keep all seqs above 200 bp enter &quot;200&quot;)
minlen=&quot;150&quot;

mkdir ${QF_reads}

for file4 in ${primer_matched}/*.fastq
    do
        echo &quot;&quot;
        echo ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        echo Quality control and removing dimer seqs
        echo input is:
        echo ${file4}

    usearch11 -fastq_filter ${file4} -fastaout &quot;${QF}/$(basename &quot;$file4&quot; .fastq).fasta&quot; -fastq_maxee ${max_ee} -fastq_minlen ${minlen}
done</code></pre>
<hr />
</div>
<div id="renaming-sequences" class="section level2">
<h2>Renaming sequences</h2>
<p>There are no usearch commands or other parameters in this step, it is simply a series of bash commands to ensure the sequence labels are compatable for downstream OTU/ZOTU clustering.</p>
<pre class="bash"><code>#!/bin/bash
QF_reads=&quot;4.quality_filtered&quot;
labeled_data=&quot;5.labeled_data&quot;

echo %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
echo Renameing sequences with &quot;&gt;barcodelabel=sample_id;sequence_id&quot;
echo %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

mkdir ${labeled_data}
mkdir working2

#*****************************************************************************************
# Step 1: Remove &quot;&gt;&quot; from start of sequence_ID

for file5 in ${QF_reads}/*.fasta
    do

    sed -e &#39;s/&gt;/&gt;barcodelabel=;/g&#39; ${file5} &gt; working2/$(basename &quot;$file5&quot; .fasta).txt
done

#*****************************************************************************************
# Step 2: Add sample_ID (should be filename) to produce &quot;&gt;barcodelabel=sample_ID;sequence_ID&quot;

for file6 in working2/*.txt
    do

    sample_id=$(basename ${file6} .txt)
    echo ${sample_id}

    sed -e &quot;s/;/${sample_id};/g&quot; ${file6} &gt; &quot;${labeled_data}/$(basename &quot;$file6&quot; .txt).fasta&quot;
done

#*****************************************************************************************
# Remove working directories

rm -r working2

##########################################################################################</code></pre>
<hr />
</div>
<div id="removing-low-abundant-sequences" class="section level2">
<h2>Removing low abundant sequences</h2>
<p><a href="https://www.drive5.com/usearch/manual/cmd_fastx_uniques.html">fastx_uniques command</a> - This command finds the set of unique sequences in an input file, also called dereplication. In this case the input is a fasta file. Sequences are compared letter-by-letter and must be identical over the full length of both sequences (substrings do not match).</p>
<p><a href="https://www.drive5.com/usearch/manual/cmd_sortbysize.html">sortbysize command</a> - This command sorts sequences by decreasing size annotation, which usually refers to the size of a cluster. In this case we also add a <code>-maxsize</code> option to specify singletons i.e. 1.</p>
<p><a href="https://www.drive5.com/usearch/manual9.2/cmd_search_exact.html">search_exact command</a> - This command searches for exact, full-length matches to a database sequence. This alogorithm is only available with usearch8, however even for large datasets the 32-bit version is fine as it does not need to store alot of memory.</p>
<pre class="bash"><code>#!/bin/bash

labeled_data=&quot;5.labeled_data&quot;
derep_dir=&quot;6.derep_data&quot;
low_abund_seqs=&quot;7.low_abund_sequences&quot;
SF=&quot;8.singleton_filtered&quot;
# Enter max replicate cluster size (eg. to remove singletons enter 1, for duplicates enter 2)
maxsize=&quot;1&quot;

echo %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
echo Removing low abundant seqs singletons per sample
echo %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
echo &quot;&quot;

# Creating directories

mkdir ${derep_dir}
mkdir ${low_abund_seqs}
mkdir ${SF}

#*****************************************************************************************
# Step 1: Dereplicating

for file7 in ${labeled_data}/*.fasta
    do

        echo &quot;&quot;
        echo ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        echo Removing singletons step 1: derep_fulllength
        echo input is:
        echo ${file7}

    usearch11 -fastx_uniques ${file7} -fastaout &quot;${derep_dir}/$(basename &quot;$file7&quot; .fasta).fasta&quot; -sizeout
done


#*****************************************************************************************
# Step 2: Filtering low abundant seqs {maxsize}

for file8 in ${derep_dir}/*.fasta
    do

        echo &quot;&quot;
        echo ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        echo Removing singletons step 2: sorting uniques
        echo input is:
        echo ${file8}

    usearch11 -sortbysize ${file8} -fastaout &quot;${low_abund_seqs}/$(basename &quot;$file8&quot; .fasta).fasta&quot; -maxsize ${maxsize}
done

#*****************************************************************************************
# Step 3: Mapping reads

for file9 in ${labeled_data}/*.fasta
    do

        echo &quot;&quot;
        echo ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        echo Removing singletons step 3: mapping reads to low abundant uniques
        echo input is:
        echo ${file9}

    usearch11 -search_exact ${file9} -db &quot;${low_abund_seqs}/$(basename &quot;$file9&quot; .fasta).fasta&quot; -strand plus -notmatched &quot;${SF}/$(basename &quot;$file9&quot; .fasta).fasta&quot;
done


#*****************************************************************************************</code></pre>
<hr />
</div>
<div id="clustering" class="section level2">
<h2>Clustering</h2>
<p>This step is where you may encounter issues with enough memory, for 1 or 2 MiSeq runs, the 32-bit versions of usearch may be okay, however any larger and you will need to use the 64-bit, paid versions. There are alternative freely available software for this step such as vsearch, and it may be worth your time exploring these options if you encounter such limits. For this case we will assume you either have a small data set or have access to the 64-bit versions for larger datasets.</p>
<p>There are two main alogorithms we recommend you do both and see what works. In my opinion I think that uparse OTUs are sufficent for 16S microbiome analysis in most cases. However if you are after more specific information (i.e. genotypes) or perhaps you targeted a more specific gene region (i.e. your primer were genus/family specific), you may want to go with ZOTUs. In short I think you should do both and make the descision of which makes biological sense in your study.</p>
<div id="uparse-algorithm-to-generate-otus" class="section level4">
<h4>UPARSE algorithm to generate OTUs</h4>
<p>The UPARSE algorithm is available in usearch 11 (and previous versions). If you have a large dataset you will need the 64-bit version as it does hold memory. Detailed documentation on this algorithm can be found <a href="https://drive5.com/usearch/manual/uparseotu_algo.html">here</a>.</p>
<p>The UPARSE-OTU algorithm constructs a set of OTU representative sequences from NGS amplicon reads. R</p>
<p>Clustering criteria The goal of UPARSE-OTU is to identify a set of OTU representative sequences (a subset of the input sequences) satisfying the following criteria.</p>
<p>All pairs of OTU sequences should have &lt; 97% pair-wise sequence identity. An OTU sequence should be the most abundant within a 97% neighborhood. Chimeric sequences should be discarded. All non-chimeric input sequences should match at least one OTU with ≥ 97% identity. UPARSE-OTU uses a greedy algorithm to find a biologically relevant solution, as follows. Since high-abundance reads are more likely to be correct amplicon sequences, and hence are more likely to be true biological sequences, UPARSE-OTU considers input sequences in order of decreasing abundance. This means that OTU centroids tend to be selected from the more abundant reads, and hence are more likely to be correct biological sequences.</p>
<p><em>Reference</em> Edgar, R.C. (2013) UPARSE: Highly accurate OTU sequences from microbial amplicon reads, Nature Methods. <a href="https://www.nature.com/articles/nmeth.2604">Pubmed:23955772, dx.doi.org/10.1038/nmeth.2604</a>.</p>
<pre class="bash"><code>#!/bin/bash

SF=&quot;8.singleton_filtered&quot;
cluster=&quot;9.cluster&quot;
uparse_otus=&quot;9a.uparse_otus&quot;

echo %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
echo UPARSE on all
echo %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

mkdir ${cluster}

   cat ${SF}/*.fasta &gt; ${cluster}/all_SF.fasta

cd ${cluster}

  usearch11 -fastx_uniques all_SF.fasta -fastaout all_SF_DR.fasta -sizeout

mkdir ${uparse_otus}
cd ${uparse_otus}

  usearch11 -cluster_otus ../all_SF_DR.fasta -otus uparse_otus.fasta -relabel OTU

  usearch11 -usearch_global ../all_SF.fasta -db uparse_otus.fasta -strand both -id 0.97 -otutabout uparse_otu_tab.txt -biomout uparse_otu_biom.biom

  usearch11 -calc_distmx uparse_otus.fasta -tabbedout uparse_otus_distmx.txt -maxdist 0.2 -termdist 0.3
  
  usearch11 -cluster_aggd uparse_otus_distmx.txt -treeout uparse_otus_clusters.tree -clusterout uparse_otus_clusters.txt \
      -id 0.80 -linkage min

cd ..</code></pre>
</div>
<div id="unoise3-algorithm-to-generate-zotus" class="section level4">
<h4>UNOISE3 algorithm to generate ZOTUs</h4>
<p>The UNOISE3 algorithm is only available in USEARCH10 onwards. If you have a large data set you will need the 64-bit version as it does hold memory. Detailed documation on this algorithm can be found <a href="https://drive5.com/usearch/manual/cmd_unoise3.html">here</a>.</p>
<p>There is a UNOISE2 option in Usearch9.2 with documentation available <a href="https://www.drive5.com/usearch/manual9.2/unoise_pipeline.html">here</a>.</p>
<p>This algorithm is specifically designed for Illumina sequences.</p>
<p>The UNOISE3 algorithm performs error-correction (denoising) on amplicon reads. It is implemented in the unoise3 command.</p>
<p>Correct biological sequences are recovered from the reads, resolving distinct sequences down to a single difference (often) or two or more differences (almost always).</p>
<p>Errors are corrected as follows: 1. Reads with sequencing and PCR point error are identified and removed. 2. Chimeras are removed.</p>
<p><em>Reference</em> Edgar, R.C. (2016), UNOISE2: Improved error-correction for Illumina 16S and ITS amplicon reads.<a href="https://www.biorxiv.org/content/early/2016/10/15/081257">http://dx.doi.org/10.1101/081257</a></p>
<pre class="bash"><code>#!/bin/bash

cluster=&quot;9.cluster&quot;
unoise_zotus=&quot;9b.unoise_zotus&quot;

echo %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
echo UNOISE on all
echo %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


mkdir ${unoise_zotus}
cd ${unoise_zotus}

  usearch11 -unoise3 ../all_SF_DR.fasta -zotus unoise_zotus.fasta -tabbedout unoise_tab.txt

  usearch11 -fastx_relabel unoise_zotus.fasta -prefix Otu -fastaout unoise_zotus_relabelled.fasta -keep_annots

  usearch11 -otutab ../all_SF_DR.fasta -zotus unoise_zotus_relabelled.fasta -otutabout unoise_otu_tab.txt -biomout unoise_otu_biom.biom -mapout unoise_map.txt -notmatched unoise_notmatched.fasta -dbmatched dbmatches.fasta -sizeout

  usearch11 -calc_distmx unoise_zotus.fasta -tabbedout unoise_zotus_distmx.txt -maxdist 0.2 -termdist 0.3
  
  usearch11 -cluster_aggd unoise_zotus_distmx.txt -treeout unoise_zotus_clusters.tree -clusterout unoise_zotus_clusters.txt \
      -id 0.80 -linkage min

cd ..
cd ..</code></pre>
<hr />
</div>
</div>
<div id="what-next" class="section level2">
<h2>What next?</h2>
<p>This is downstream analysis that can done with usearch. For example, <a href="https://drive5.com/usearch/manual/cmds_taxonomy.html">taxonomy assignment</a> and <a href="https://drive5.com/usearch/manual/cmds_div.html">diversity analysis</a>.</p>
<p>However, at present there are more robust taxonomy assignment options available (such as that in QIIME2), that are easier to use. With respect to diversity analysis, QIIME2 and R packages such as Phyloseq and Microbiome offer much more flexibility, robustness and are more aesthectically pleasing.</p>
</div>

<p>
<br></br>Vector and Waterborne Pathogens Research Group. 2018. S Egan.
</p>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
